File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumIntervaloAlertaEmail.js
var EnumIntervaloAlertaEmailHelper = function() {
    this.Todos = "";
    this.Dia = 1;
    this.Mes = 2;
    this.Semana = 3;
    this.Ano = 4;
};


EnumIntervaloAlertaEmailHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Dia", value: this.Dia },
            { text: "Mês", value: this.Mes },
            { text: "Semana", value: this.Semana },
            { text: "Ano", value: this.Ano }
        ];
    },
    obterOpcoesPesquisa: function () {
        return [{ text: Localization.Resources.Gerais.Geral.Todos, value: this.Todos }].concat(this.obterOpcoes());
    }
};

var EnumIntervaloAlertaEmail = Object.freeze(new EnumIntervaloAlertaEmailHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumIntervaloAlertaEmail.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumIntervaloAlertaEmail
    {
        Dia = 1,
        Semana = 2,
        Mes = 3,
        Ano = 4,
    }

    public static class EnumIntervaloAlertaEmailHelper
    {
        public static string ObterDescricao(this EnumIntervaloAlertaEmail periodicidade)
        {
            switch (periodicidade)
            {
                case EnumIntervaloAlertaEmail.Dia: return "Dia";
                case EnumIntervaloAlertaEmail.Semana: return "Semana";
                case EnumIntervaloAlertaEmail.Mes: return "Mes";
                case EnumIntervaloAlertaEmail.Ano: return "Ano";
                default: return string.Empty;
            }
        }
    }
}
